packages/dd-trace/src/config.js

```diff
@@ -211,33 +211,27 @@ function propagationStyle (key, option, defaultValue) {
 }
 
 class Config {
-  constructor (options) {
-    options = options || {}
+  constructor (options = {}) {
     options = this.options = {
       ...options,
       appsec: options.appsec != null ? options.appsec : options.experimental?.appsec,
       iast: options.iast != null ? options.iast : options.experimental?.iast
     }
 
-    checkIfBothOtelAndDdEnvVarSet()
```

Comment:
I moved this down below the update to the logger to be true to the "Configure the logger first so it can be used to warn about other configs" comment below since the call to `checkIfBothOtelAndDdEnvVarSet` might try to log some warnings.